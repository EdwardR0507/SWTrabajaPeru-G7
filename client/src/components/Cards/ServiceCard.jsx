import React, { useState, useEffect } from "react";
import { withStyles } from "@material-ui/styles";
import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import CardMedia from "@material-ui/core/CardMedia";
import CardContent from "@material-ui/core/CardContent";
import CardActions from "@material-ui/core/CardActions";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import IconButton from "@material-ui/core/IconButton";
import ShareIcon from "@material-ui/icons/Share";
import Avatar from "@material-ui/core/Avatar";
import theme from "../../themes/themes";

const StyledCard = withStyles({
  root: {
    width: "25%",
    margin: "20px auto",
  },
})(Card);

const StyledCardMedia = withStyles({
  root: {
    height: "0",
    paddingTop: "41%",
  },
})(CardMedia);

const StyledCardActions = withStyles({
  root: {
    display: "flex",
    justifyContent: "space-between",
  },
})(CardActions);

const StyledButton = withStyles({
  root: {
    color: theme.palette.primary.main,
  },
})(Button);

const StyledBody2 = withStyles({
  root: {
    color: theme.cardLetter.primary.main,
  },
})(Typography);

const StyledCaption = withStyles({
  root: {
    marginRight: "5%",
    color: theme.cardLetter.primary.main,
  },
})(Typography);

const ServiceCard = (props) => {
  const [service, setService] = useState();

  useEffect(()=>{
    setService(props.service)
  }, [])

  return service ? (
    <StyledCard>
      {/*Cambiar los datos por informaciónde la bd*/}
      <CardHeader
        avatar={<Avatar></Avatar>}
        title={service.cat_nombre}
        subheader={service.us_nombres}
        action={
          <IconButton aria-label="share">
            <ShareIcon />
          </IconButton>
        }
      />
      <StyledCardMedia image="src/assets/CardTest.jpeg" />
      <CardContent>
        <StyledBody2 variant="body2">
          {service.ser_descripcion}
        </StyledBody2>
      </CardContent>
      <StyledCardActions>
        <StyledButton>Ver Más</StyledButton>
      </StyledCardActions>
    </StyledCard>
  ):(
    <div>Cargando...</div>
  );
};

export default ServiceCard;
