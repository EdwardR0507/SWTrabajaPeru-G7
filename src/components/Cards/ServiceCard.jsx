import React, { useState, useEffect } from "react";
import { useHistory } from "react-router";
import { withStyles } from "@material-ui/styles";
import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import CardMedia from "@material-ui/core/CardMedia";
import CardContent from "@material-ui/core/CardContent";
import CardActions from "@material-ui/core/CardActions";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import IconButton from "@material-ui/core/IconButton";
import ShareIcon from "@material-ui/icons/Share";
import Avatar from "@material-ui/core/Avatar";
import DialogLogin from "../Dialog/DialogLogin";
import theme from "../../themes/themes";
import image from "../../assets/services.jpg";
import imageWorker from "../../assets/worker.jpg";
const StyledCard = withStyles({
  root: {
    width: "300px",
    margin: "20px auto",
  },
})(Card);

const StyledCardMedia = withStyles({
  root: {
    height: "0",
    paddingTop: "41%",
  },
})(CardMedia);

const StyledCardActions = withStyles({
  root: {
    display: "flex",
    justifyContent: "space-between",
  },
})(CardActions);

const StyledButton = withStyles({
  root: {
    color: theme.palette.primary.main,
  },
})(Button);

const StyledBody2 = withStyles({
  root: {
    color: theme.cardLetter.primary.main,
  },
})(Typography);

const StyledCaption = withStyles({
  root: {
    marginRight: "5%",
    color: theme.cardLetter.primary.main,
  },
})(Typography);

const ServiceCard = (props) => {
  const [service, setService] = useState();
  const history = useHistory();

  useEffect(() => {
    setService(props.service);
  }, []);

  return service ? (
    <StyledCard>
      {/*Cambiar los datos por informaciónde la bd*/}
      <CardHeader
        avatar={<Avatar src={imageWorker}></Avatar>}
        title={service.cat_nombre}
        subheader={service.us_nombres}
        action={
          <IconButton aria-label="share">
            <ShareIcon />
          </IconButton>
        }
      />
      <StyledCardMedia image={image} />
      <CardContent>
        <StyledBody2 variant="body2">{service.ser_descripcion}</StyledBody2>
      </CardContent>
      <StyledCardActions>
        {props.token ? (
          <StyledButton
            role="redirect"
            onClick={() => {
              history.push({
                pathname: "/serviceDetails",
                search: `user=${service.us_correo.split("@")}&?service=${
                  service.cat_id
                }`,
                state: {
                  token: props.token,
                  us_id: service.us_id,
                  cat_id: service.cat_id,
                },
              });
            }}
          >
            Ver Más
          </StyledButton>
        ) : (
          <DialogLogin />
        )}
      </StyledCardActions>
    </StyledCard>
  ) : (
    <div>Cargando...</div>
  );
};

export default ServiceCard;
