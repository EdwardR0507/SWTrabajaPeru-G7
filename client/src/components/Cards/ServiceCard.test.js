/*Importamos las librerias principales*/
import React from "react";
import { shallow } from "enzyme";
import ServiceCard from "./ServiceCard";
/*Renderizado del componente ServiceCard*/
describe("<ServiceCard></ServiceCard>", () => {
    it('render "ServiceCard"', () => {
        const wrapper = shallow(<ServiceCard />);
        expect(wrapper).toMatchSnapshot()
      });
});
