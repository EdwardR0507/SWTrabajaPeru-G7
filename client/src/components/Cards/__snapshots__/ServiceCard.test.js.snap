// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServiceCard></ServiceCard> render "ServiceCard" 1`] = `ShallowWrapper {}`;
